deckhouse-controller/pkg/controller/module-controllers/utils/utils.go

```diff
@@ -220,3 +233,175 @@ func GetClusterUUID(ctx context.Context, cli client.Client) string {
 	// generate a random UUID if the key is missing
 	return uuid.Must(uuid.NewV4()).String()
 }
+
+// EnableModule deletes old symlinks and creates a new one
+func EnableModule(downloadedModulesDir, oldSymlinkPath, newSymlinkPath, modulePath string) error {
+	// delete the old module symlink with diff version if exists
+	if oldSymlinkPath != "" {
+		if _, err := os.Lstat(oldSymlinkPath); err == nil {
+			if err = os.Remove(oldSymlinkPath); err != nil {
+				return fmt.Errorf("delete the '%s' old symlink: %w", oldSymlinkPath, err)
+			}
+		}
+	}
+
+	// delete the new module symlink
+	if _, err := os.Lstat(newSymlinkPath); err == nil {
+		if err = os.Remove(newSymlinkPath); err != nil {
+			return fmt.Errorf("delete the '%s' new symlink: %w", newSymlinkPath, err)
+		}
+	}
+
+	// make absolute path for versioned module
+	moduleAbsPath := filepath.Join(downloadedModulesDir, strings.TrimPrefix(modulePath, "../"))
+	// check that module exists on a disk
+	if _, err := os.Stat(moduleAbsPath); os.IsNotExist(err) {
+		return fmt.Errorf("the '%s' module absolute path not found", moduleAbsPath)
```

Comment:
looks like place for a constant Err

Reply:
Why ?

Reply:
i thinking about handling this error in upper layer. not necessary